src/Microsoft.Identity.Web/AuthorityHelpers.cs

```diff
@@ -8,52 +8,28 @@ namespace Microsoft.Identity.Web
 {
     internal static class AuthorityHelpers
     {
-        internal static bool IsV2Authority(string authority)
-        {
-            if (string.IsNullOrEmpty(authority))
-                return false;
-
-            return authority.EndsWith("/v2.0");
-        }
-
         internal static string BuildAuthority(MicrosoftIdentityOptions options)
         {
-            if (options == null)
-                return null;
-
-            // Cannot build authority without AAD Instance
-            if (string.IsNullOrWhiteSpace(options.Instance))
-                return null;
-
             var baseUri = new Uri(options.Instance);
             var pathBase = baseUri.PathAndQuery.TrimEnd('/');
             var domain = options.Domain;
             var tenantId = options.TenantId;
 
-            // If there are user flows, then it must build a B2C authority 
-            if (!string.IsNullOrWhiteSpace(options.DefaultUserFlow))
+            if (options.IsB2C)
             {
-                // Cannot build B2C authority without domain
-                if (string.IsNullOrWhiteSpace(domain))
-                    return null;
-
                 var userFlow = options.DefaultUserFlow;
                 return new Uri(baseUri, new PathString($"{pathBase}/{domain}/{userFlow}/v2.0")).ToString();
             }
-            else
+            else // Create an AAD authority
```

Comment:
If it's not B2C, it can be ADFS.

Reply:
@bgavrilMS : do you think that Microsoft.Identity.Web should support ADFS?

Reply:
I think you should at least throw a "not supported" exception or smth.